plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -626,4 +632,34 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
         }
         return Optional.empty();
     }
+
+    private void validateSchemaTable(SchemaTableName schemaTableName)
+    {
+        if (metastore.getDatabase(schemaTableName.getSchemaName()).isEmpty()) {
+            throw new TrinoException(ICEBERG_INVALID_METADATA, format("Schema '%s' does not exist", schemaTableName.getSchemaName()));
+        }
+        if (metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName()).isPresent()) {
+            throw new TrinoException(ICEBERG_INVALID_METADATA, format("Table '%s' already exists in schema '%s'", schemaTableName.getTableName(), schemaTableName.getSchemaName()));
+        }
+    }
+
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation, Optional<String> metadataFileName)
+    {
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+
+        validateSchemaTable(schemaTableName);
```

Comment:
Inline `validateSchemaTable` content in the method.

Reasoning behind the request: `validateSchemaTable` is a rather generic name.

Reply:
Addressed! Created `tableExists` and `schemaExists` and inline the call.